build.sh

```diff
@@ -76,6 +76,42 @@ function clean_image {
   fi
 }
 
+# Pull image based on FROM, before we build our own.
+function pull_image {
+  local dockerfile="$1"
+
+  # Get image_name from Dockerfile before pulling.
+  while read -r line; do
+    if ! grep -q "^FROM" <<< "$line"; then
+      continue
+    fi
+
+    image_name=$(echo "$line" | cut -d ' ' -f2)
+    # In case FROM scratch is defined, skip it
+    if grep -q "scratch" <<< "$image_name"; then
+        continue
+    fi
+    # In case FROM ubi8 as ubi-base is mentioned
+    # pull ubi8 image
+    if grep -q " as " <<< "$line"; then
+        image_name=$(echo "$line" | cut -d ' ' -f2)
+    fi
+    echo "-> Pulling image $image_name before building image from $dockerfile."
+    # Sometimes in Fedora case it fails with HTTP 50X
+    # Check if the image is available locally and try to pull it if it is not
+    if docker images "$image_name" &>/dev/null; then
```

Comment:
Trying this out for an image that does not exist, the docker/podman command exits with return value 0 either way.

Reply:
Some example is here: https://github.com/VonC/b2d/blob/2d671bbe1311b63b24e7b8b495e3055dc5c5f97b/apache/build#L6-L8

Reply:
Yes, that would work. Thanks!